src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,305 @@
+            writeTaskToFile(pw);
+
+        }  catch (FileNotFoundException e) {
+            File dir = new File("./data");
+            if (!dir.isDirectory()) {
+                System.out.println("Data directory not found. Creating new data directory");
+                Path path = Paths.get("./data");
+                try {
+                    Files.createDirectories(path);
+                } catch (IOException f) {
+                    System.out.println("Failed to create directory");
+                }
+            }
+            if (!filePath.isFile()) {
+                System.out.println("File not found. Creating new text file");
+                Path fileLoc = Paths.get("./data/duke.txt");
+                try {
+                    Files.createFile(fileLoc);
+                } catch (IOException f) {
+                    System.out.println("Failed to create file");
+                }
+            }
+
+        }
+    }
+
+    public static void writeTaskToFile(PrintWriter pw) {
+        for (int i = 0; i < tasks.size(); ++i) {
+            pw.println(tasks.get(i).printToFile());
+        }
+        pw.close();
+    }
+    public static void initialiseTaskList(String line) {
+        String[] task = line.split("\\||-");
+
+        switch(task[0].trim()) {
+        case "T":
+            addTodoTask(task[2].trim());
+            break;
+        case "D":
+            addDeadlineTask(task[2].trim(), task[3].trim());
+            break;
+        case "E":
+            addEventTask(task[2].trim(), task[3].trim(), task[4].trim());
+            break;
+        }
+
+        if (task[1].contains("X")) {
+            try {
+                markTaskDone(tasks.size()-1);
+            } catch (NumberFormatException e) {
+                System.out.println("☹ OOPS!!! Task number should be an integer.");
+                printDivider();
+            } catch (InvalidTaskNumberException e) {
+                System.out.println("☹ OOPS!!! The task specified does not exist in the task list.");
+                printDivider();
+            } catch (IndexOutOfBoundsException e) {
+                System.out.println("☹ OOPS!!! The description of 'mark' cannot be empty.");
+                printDivider();
+            }
+        }
+    }
+
+    public static void printDivider () {
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void printStartMessage() {
+        String logo = " ____\n"
+                + "|    \\ ___ _ _ ___\n"
+                + "|  |  | .'| | | -_|\n"
+                + "|____/|__,|\\_/|___|\n";
+
+        printDivider();
+        System.out.println("List Summary:");
+        printTaskList();
+        System.out.print(logo);
+        System.out.println("Hi, I'm Dave!\n"
+                + "What can I do for you?");
+        printDivider();
+    }
+
+    public static void processCommand (String input) throws CommandNotRecognisedException {

```

Comment:
This method is a bit long. You might consider create a new methods for each case. For example in case Command.COMMAND_MARK, you can make a method to validate the command and call it in the case block. So that the processCommand method will be shorter and prettier. One more thing is that the space between method name and parameter can be removed